src/handwritten/system/_collections/physical_channel_collection.py

```diff
@@ -59,9 +59,26 @@ def __getitem__(self, index):
         if isinstance(index, int):
             return _PhysicalChannelAlternateConstructor(self.channel_names[index], self._interpreter)
         elif isinstance(index, slice):
-            return _PhysicalChannelAlternateConstructor(self.channel_names[index], self._interpreter)
+            return [_PhysicalChannelAlternateConstructor(channel, self._interpreter) for channel in self.channel_names[index]]
         elif isinstance(index, str):
-            return _PhysicalChannelAlternateConstructor(f'{self._name}/{index}', self._interpreter)
+            requested_channels = unflatten_channel_string(index)
+            all_channels = self.channel_names
+            # Validate the channel names we were provided
+            channels_to_use = []
+            for channel in requested_channels:
+                if channel in all_channels:
+                    channels_to_use.append(channel)
+                else:
+                    # The channel may have been unqualified, so we'll try to qualify it
+                    qualified_channel = f'{self._name}/{channel}'
+                    if qualified_channel in all_channels:
+                        channels_to_use.append(qualified_channel)
+                    else:
+                        raise KeyError(f'"{channel}" is not a valid channel name.')
```

Comment:
I think `self.channel_names` only contains external channels, so this validation prevents accessing internal channels.

We may need to add internal C APIs to do this validation anyway: AB#2663510

For now, I would check `channel.startswith(f"{self._name}")` to see if it's fully qualified.

Reply:
I think I'll remove the validation from both these collections, and only detect whether or not we need to add a device name.